src/main/java/com/hartwig/platinum/kubernetes/KubernetesClientProxy.java

```diff
@@ -0,0 +1,61 @@
+package com.hartwig.platinum.kubernetes;
+
+import static java.util.List.of;
+
+import javax.inject.Provider;
+
+import com.hartwig.platinum.config.GcpConfiguration;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import io.fabric8.kubernetes.api.model.ConfigMap;
+import io.fabric8.kubernetes.api.model.ConfigMapList;
+import io.fabric8.kubernetes.api.model.Secret;
+import io.fabric8.kubernetes.api.model.SecretList;
+import io.fabric8.kubernetes.client.KubernetesClient;
+import io.fabric8.kubernetes.client.dsl.MixedOperation;
+import io.fabric8.kubernetes.client.dsl.Resource;
+
+public class KubernetesClientProxy {
+    private static final Logger LOGGER = LoggerFactory.getLogger(KubernetesClientProxy.class);
+    private final String clusterName;
+    private final String region;
+    private final String project;
+    private Provider<KubernetesClient> kubernetesClientProvider;
```

Comment:
why a lazy init provider? Could it not just wrap the client? If not I guess can be final?